test/retry.js

```diff
@@ -0,0 +1,31 @@
+import test from 'ava';
+import got from '../';
+import {createServer} from './_server';
+
+const s = createServer();
+
+s.on('/long', () => {});
+
+let knocks = 0;
+s.on('/knock-twice', (req, res) => {
+	if (knocks++ === 1) {
+		res.end('who`s there?');
+	}
+});
+
+test.before('retry - setup', t => {
+	s.listen(s.port, () => t.end());
+});
+
+test('retry - timeout errors', t => {
+	got(s.url + '/knock-twice', {timeout: 1000}, (err, data) => {
+		t.ifError(err);
+		t.is(data, 'who`s there?');
+		t.end();
+	});
```

Comment:
> I think we should start using the promise interface as default for new tests. It makes the test a lot simpler.

Yup, it looks way better. I guess this would be in a separate PR.